pkg/stanza/fileconsumer/internal/reader/factory.go

```diff
@@ -28,7 +29,7 @@ const (
 )
 
 type Factory struct {
-	*zap.SugaredLogger
+	component.TelemetrySettings
```

Comment:
Do we need this to be a public field?

Reply:
Unfortunately yes, for now at least, because we create the factory manually. (Rather than having a builder or factory for the factory). However, it is in an very internal package of `fileconsumer` so it's not actually increasing our surface area.

Reply:
Makes sense, thanks for clarifying